tests/unittests/test_datasource/test_ec2.py

```diff
@@ -152,19 +154,19 @@
 }
 
 NIC2_MD = {
-    "device_number": "1",
```

Comment:
The [AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html) and the examples with provenance all use "-" instead of "_", and on an instance I just launched I see this in the logs:

```
Read from http://169.254.169.254/2018-09-24/meta-data/network/interfaces/macs/16:45:df:fa:16:8f/device-number
```

We didn't notice this before because before my change the code would just log if it was missing expected attributes (either because they aren't expected to be present except in specific configurations, or because we're being overly cautious, I'm not sure which).

I _think_ this is a copy/paste from `cloud-init query`'s output instead of from the IMDS itself, perhaps @blackboxsw can weigh in?

Reply:
Yes, I'm happy with the docs; I suspect it was a local transformation;